### PR TITLE
Improve `get_device` error message when ChainerX is not available

### LINKED ISSUE #7350: Improve error message when ChainerX is unavailable
When a user attempt to use `cuda:0` device, ChainerX says the device specifier is invalid if `chainerx.is_available()` is `False`. It seems a bit difficult to deduce the actual problem from the message.

```
$ python train_mnist.py -d cuda:0
Traceback (most recent call last):
  File "train_mnist.py", line 134, in <module>
    main()
  File "train_mnist.py", line 56, in main
    device = chainer.get_device(args.device)
  File "/path/to/chainer/chainer/backend.py", line 157, in get_device
    raise ValueError('Invalid device specifier: {}'.format(device_spec))
ValueError: Invalid device specifier: cuda:0
```

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7350

### CODE REVIEW

**Comment by gwtnb** [comment]
I noticed that the incorrect message is show not only when ChainerX is unavailable, but also when `cuda:n` device is unavailable (i.e. ChainerX is not built with CUDA or `n` is too large). In the situation, ChainerX should tell the user that ChainerX is not built with CUDA or that there is no such device.

File: chainer/backend.py [hunk 1]
```diff
@@ -154,7 +154,16 @@ def get_device(device_spec):
...
+            if not chainerx.is_available():
+                raise RuntimeError(
+                    'Tried to parse ChainerX device specifier \'{}\', '
+                    'but ChainerX is not available. '
+                    'Note that device specifiers without \'@\' prefix are '
+                    'assumed to be ChainerX device '
+                    'specifiers.'.format(device_spec))
             return _chainerx.ChainerxDevice(chainerx.get_device(device_spec))
 
     raise ValueError('Invalid device specifier: {}'.format(device_spec))
```
Comment (hvy) [inline chainer/backend.py]: Is a `TypeError` better suited here? Maybe not within the scope of this PR.

**Comment by niboshi** [comment]
@gwtnb Hmm, is that true?
>I noticed that the incorrect message is shown not only when ChainerX is unavailable, but also when `cuda:n` device is unavailable (i.e. ChainerX is not built with CUDA or `n` is too large). In the situation, ChainerX should tell the user that ChainerX is not built with CUDA or that there is no such device.

I'm getting a comprehensible error:
```
>>> chainer.get_device('cuda:100')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/niboshi/w/repos/chainer/chainer/backend.py", line 158, in get_device
    return _chainerx.ChainerxDevice(chainerx.get_device(device_spec))
IndexError: The index number (= 100) is not less than the device count (= 4)
```

File: chainer/backend.py (hunk 1 above)
Comment (niboshi) [reply chainer/backend.py]: I don't know which is better. Surely it's out of scope of this PR.

**Comment by hvy** [comment]
Is it a matter of catching the ChainerX errors?

**Comment by niboshi** [comment]
What do you mean? Do you mean we should wrap the `IndexError` and reraise with a message which mentions ChainerX?
I think the current message is descriptive enough.

**Comment by hvy** [comment]
Yes, that's what I meant. I am not sure if that's what @gwtnb meant.

**Comment by gwtnb** [comment]
Sorry, this is due to a problem on my environment. https://github.com/chainer/chainer/pull/7401#issuecomment-499028640

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM, failure is unrelated.

File: chainer/backend.py (hunk 1 above)
Comment (hvy) [reply chainer/backend.py]: Created a PR. Could be open for discussion. https://github.com/chainer/chainer/pull/7421

**Pull request merged**